node/bft/src/sync/mod.rs

```diff
@@ -237,14 +262,40 @@ impl<N: Network> Sync<N> {
         let self_ = self.clone();
         self.spawn(async move {
             while let Some((peer_ip, certificate_response)) = rx_certificate_response.recv().await {
-                self_.finish_certificate_request(peer_ip, certificate_response)
+                self_.finish_certificate_request(peer_ip, certificate_response);
             }
         });
 
         Ok(())
     }
 }
 
+// Callbacks used when receiving messages from the Gateway
+impl<N: Network> Sync<N> {
+    /// We received a block response and can (possibly) advance synchronization.
+    async fn advance_with_sync_blocks(&self, peer_ip: SocketAddr, blocks: Vec<Block<N>>) -> Result<()> {
+        // Verify that the response is valid.
+        self.block_sync.insert_block_responses(peer_ip, blocks)?;
```

Comment:
Is there a reason we are avoiding the previous abstraction of `self_.block_sync.process_block_response(peer_ip, blocks)` and instead calling `insert_block_responses` directly?